app/scripts/models/account.js

```diff
@@ -104,27 +104,20 @@ define(function (require, exports, module) {
 
     // Hydrate the account
     fetch: function () {
-      var self = this;
-      var promise = p();
-
-      if (! self.get('sessionToken')) {
-        return promise;
+      if (! this.get('sessionToken') || this.get('verified')) {
+        return p();
       }
 
       // upgrade the credentials with verified state
-      if (! self.get('verified')) {
-        promise = self.isVerified()
```

Comment:
I replaced the call to `isVerified` with a direct call to `sessionStatus` and simplified all the promise related stuff.